docs/source/developers/java/building.rst

```diff
@@ -171,11 +171,31 @@ IDE Configuration
 IntelliJ
 --------
 
-To start working on Arrow in IntelliJ, just open the `java/`
-subdirectory of the Arrow repository.
-
+To start working on Arrow in IntelliJ: build the project once from the command
+line using ``mvn clean install``. Then open the ``java/`` subdirectory of the
+Arrow repository, and update the following settings:
+
+* In the Files tool window, find the path ``vector/target/generated-sources``,
+  right click the directory, and select Mark Directory as > Generated Sources
+  Root. There is no need to mark other generated sources directories, as only
+  the ``vector`` module generates sources.
 * For JDK 8, disable the ``error-prone`` profile to build the project successfully.
-* For JDK 11, the project should build successfully with the default profiles.
+* For JDK 11, due to an `IntelliJ bug
+  <https://youtrack.jetbrains.com/issue/IDEA-201168>`__, you must go into
+  Settings > Build, Execution, Deployment > Compiler > Java Compiler and disable
+  "Use '--release' option for cross-compilation (Java 9 and later)". Otherwise
+  you will get an error like "package sun.misc does not exist".
+* You may need to disable the ``linux-netty-native`` profile in the Maven tool
```

Comment:
You may need to disable the ``linux-netty-native`` or ``mac-netty-native`` profile in the Maven tool